### PR TITLE
🔐 Add Login page

### LINKED ISSUE #161: Add login for companies with encrypted database

### PR DESCRIPTION
**Pull request by bwnstck**
# Description

Fixes #161 

## Type of change

- [x] New feature (non-breaking change which adds functionality)

### CODE REVIEW

File: client/src/components/helpers/CenterContent.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import styled from "styled-components/macro";
+
+export const CenterContent = styled.div`
+  display: grid;
+  grid-template-rows: 100%;
```
Comment (BelaM80) [inline client/src/components/helpers/CenterContent.js]: What´s the use of the grid-template here?

File: client/src/pages/Login.js [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+              <img src={Micha} alt="Micha" />
+            </Illustrations>
+          </ContentWrapper>
+        </CenterContent>
+      </Wrapper>
+    </div>
+  );
+};
+
+const Title = styled.h1`
```
Comment (BelaM80) [inline client/src/pages/Login.js:87]: Is this the only title you have? If you have more, wouldn´t it make sense to define it in globalStyles? And if you have only this one, you could define the style in the ContentWrapper component.

File: client/src/pages/Login.js [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+const Illustrations = styled.div`
+  display: flex;
+  justify-content: center;
+  margin: 2rem auto 1rem;
+  > * {
+    height: clamp(30px, 5vw, 70px);
+    margin: auto 1rem;
+  }
+`;
+const Subtitle = styled.h2`
```
Comment (BelaM80) [inline client/src/pages/Login.js:133]: Same as in Title.

File: client/src/assets/login.svg [hunk 4]
```diff
@@ -0,0 +1,51 @@
+<?xml version="1.0" encoding="iso-8859-1"?>
+<!-- Generator: Adobe Illustrator 19.0.0, SVG Export Plug-In . SVG Version: 6.00 Build 0)  -->
+<svg version="1.1" id="Capa_1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" x="0px" y="0px"
+	 viewBox="0 0 511.996 511.996" style="enable-background:new 0 0 511.996 511.996;" xml:space="preserve">
```
Comment (lmachens) [inline client/src/assets/login.svg]: You can remove unused properties (clean up svgs)

File: client/src/assets/login.svg [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+<g>
+</g>
+<g>
+</g>
+<g>
+</g>
+<g>
+</g>
+<g>
+</g>
```
Comment (lmachens) [inline client/src/assets/login.svg]:
```suggestion
```

File: client/src/components/Button.js [hunk 6]
```diff
@@ -45,10 +45,19 @@ const designs = {
...
 };
 
-const Button = ({ primary, type, design, category, label, ...props }) => {
+const Button = ({
+  primary,
+  type,
+  design,
+  category,
+  label,
+  children,
```
Comment (lmachens) [inline client/src/components/Button.js:54]: Can you combine `label` and `children` into a single property?

File: client/src/components/helpers/CenterContent.js [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+  place-items: center;
+  height: 100%;
+`;
+
+// display: flex;
+// flex-direction: column;
+// align-items: center;
+// justify-content: center;
+// height: 100%;
```
Comment (lmachens) [inline client/src/components/helpers/CenterContent.js]:
```suggestion
```

File: client/src/components/helpers/Wrapper.js [hunk 8]
```diff
@@ -1,6 +1,7 @@
 import styled from "styled-components/macro";
 
 export const ContentWrapper = styled.div`
+  position: relative;
```
Comment (BelaM80) [inline client/src/components/helpers/Wrapper.js:4]: I think when you use position relative, you need to use position: absolute in the container around..

**Review by BelaM80 — APPROVED** [review]

File: client/src/pages/Launch.js [hunk 9]
```diff
@@ -1,15 +1,65 @@
...
+          src={logoDark}
+          alt="Logo Dark"
+          onClick={() => history.push("/menu")}
+        />
+      </LogoContainer>
+      <EnterContainer>
+        <Button
+          label="login"
+          design="menu"
+          onClick={() => history.push("/login")}
```
Comment (lmachens) [inline client/src/pages/Launch.js]: Better use a link instead of a button if you use it for navigation only.

**Review by lmachens — COMMENTED** [review]
gj 🎉

regarding the height issue:
https://stackoverflow.com/questions/8468066/child-inside-parent-with-min-height-100-not-inheriting-height
https://bugs.webkit.org/show_bug.cgi?id=26559

File: client/src/pages/Login.js (hunk 2 above)
Comment (bwnstck) [reply client/src/pages/Login.js:87]: i import the wrapper so its kind of hard to use it to style them

**Review by arassaria — APPROVED** [review]
Looks 🤓 🦄 🚀  to me

**Pull request merged**